battery_health.h

```diff
@@ -0,0 +1,127 @@
+#ifndef __BATTERY_HEALTH_H_
+#define __BATTERY_HEALTH_H_
+
+#include <vector>
+#include "email.h"
+#include "controller.h"
+
+const std::string TEMPERATURE_ALERT =  "Hi, the temperature is ";
+const std::vector<std::string> batteryBreach = {"normal", "too low", "too high"};
+
+typedef enum {
+  NORMAL = 0,
+  TOO_LOW,
+  TOO_HIGH
+} BreachType;
+
+class BatteryCoolingType {
+
+  public:
+
+    BatteryCoolingType(){}
+    ~BatteryCoolingType(){}
+    virtual int getLowerLimit() = 0;
+    virtual int getUpperLimit() = 0;
+    BreachType inferBreach(double value);
+};
+
+class PassiveCooling : public BatteryCoolingType {
+
+  protected:
+
+    int lowerLimit;
+    int upperLimit;
+
+  public:
+
+    PassiveCooling(){
+      lowerLimit = 0;
+      upperLimit = 35;
+    }
+    ~PassiveCooling(){}
+    int getLowerLimit()override;
+    int getUpperLimit()override;
+};
+
+class HighActiveCooling : public BatteryCoolingType {
+
+  protected:
+
+    int lowerLimit;
+    int upperLimit;
+
+  public:
+
+    HighActiveCooling(){
+      lowerLimit = 0;
+      upperLimit = 45;
+    }
+    ~HighActiveCooling(){}
+    int getLowerLimit()override;
+    int getUpperLimit()override;
+};
+
+class MediumActiveCooling : public BatteryCoolingType {
+
+  protected:
+
+    int lowerLimit;
+    int upperLimit;
+
+  public:
+
+    MediumActiveCooling(){
+      lowerLimit = 0;
+      upperLimit = 40;
+    }
+    ~MediumActiveCooling(){}
+    int getLowerLimit()override;

```

Comment:
if the behaviour of this method is to return lowerlimit only , without any processing being done on the class member, then override is not necessary if the method moves to base class.